vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
OK, so please decide with @sargun and @drewkerrigan whether we want to support these users or not - I am totally fine with both (== closing/merging the PR). 

Thanks for doing the review !